src/main/java/iudx/rs/proxy/authenticator/JwtAuthenticationServiceImpl.java

```diff
@@ -1,402 +1,292 @@
-  final String audience;
-  final CacheService cache;
-  final Api apis;
-  final String catBasePath;
-  // resourceGroupCache will contain ACL info about all resource group in a resource server
-  Cache<String, String> resourceGroupCache = CacheBuilder.newBuilder().maximumSize(1000)
-      .expireAfterAccess(Constants.CACHE_TIMEOUT_AMOUNT, TimeUnit.MINUTES).build();
-
-  // resourceIdCache will contain info about resources available(& their ACL) in resource server.
-  Cache<String, String> resourceIdCache = CacheBuilder.newBuilder().maximumSize(1000)
-      .expireAfterAccess(Constants.CACHE_TIMEOUT_AMOUNT, TimeUnit.MINUTES).build();
+    private static final Logger LOGGER = LogManager.getLogger(JwtAuthenticationServiceImpl.class);
+    final JWTAuth jwtAuth;
+    final String audience;
+    final CacheService cache;
+    final Api apis;
 
 
-  JwtAuthenticationServiceImpl(Vertx vertx, final JWTAuth jwtAuth, final JsonObject config,
-                               final CacheService cacheService, final Api apis) {
-    this.jwtAuth = jwtAuth;
-    this.audience = config.getString("audience");
-    this.host = config.getString("catServerHost");
-    this.port = config.getInteger("catServerPort");
-    this.catBasePath = config.getString("dxCatalogueBasePath");
-    this.path = catBasePath + CAT_SEARCH_PATH;
-    this.apis = apis;
-    WebClientOptions options = new WebClientOptions();
-    options.setTrustAll(true).setVerifyHost(false).setSsl(true);
-    catWebClient = WebClient.create(vertx, options);
-    this.cache = cacheService;
-  }
-
-  @Override
-  public AuthenticationService tokenIntrospect(JsonObject request, JsonObject authenticationInfo,
-                                               JwtData jwtData,
-                                               Handler<AsyncResult<JsonObject>> handler) {
-    LOGGER.debug("tokenIntrospect() started");
-    String id = authenticationInfo.getString("id");
-    ResultContainer result = new ResultContainer();
-    result.jwtData = jwtData;
-    String endPoint = authenticationInfo.getString("apiEndpoint");
-    boolean skipResourceIdCheck =
-            endPoint.equalsIgnoreCase(apis.getAsyncStatusEndpoint());
+    JwtAuthenticationServiceImpl(Vertx vertx, final JWTAuth jwtAuth, final JsonObject config,
+                                 final CacheService cacheService, final Api apis) {
+        this.jwtAuth = jwtAuth;
+        this.audience = config.getString("audience");
+        this.apis = apis;
+        this.cache = cacheService;
+    }
 
+    @Override
+    public AuthenticationService tokenIntrospect(JsonObject request, JsonObject authenticationInfo,
+                                                 JwtData jwtData,
+                                                 Handler<AsyncResult<JsonObject>> handler) {
+        LOGGER.info("tokenIntrospect() started");
+        String id = authenticationInfo.getString("id");
+        String ppbNumber = authenticationInfo.getString(PPB_NUMBER);
```

Comment:
Can be null in case of non-personal data, assign default value using `authenticationInfo.getString(PPB_NUMBER, "");`

Reply:
done.